### DISCUSSION TITLE
issue: isDirty form state set to true even if i manually reset it

### DISCUSSION THREAD

**Discussion by kkatsi** [react-hook-form/react-hook-form#12083] [Q&A] [answered]
### Version Number

7.52.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-isdirty-state-d6mgz2?file=%2Fsrc%2FApp.tsx%3A1%2C1-34%2C1

### Steps to reproduce

1. Open console (isDirty = false)
2. Go to input and type a value (isDirty = true)
3. Click on submit button 
4. See on console that isDirty sets to 'false' after submit and immediately becoming 'true' again 

### Expected behaviour

According to the docs and several similar issues i read, i expect to run
`reset({}, {keepValues: true})` 
resetting the isDirty state and keeping the user values.
Am i missing something?

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
try this `reset(getValues(), { keepDirty: true });` 

- update defaultValues 
- update values
- keep the dirty form state
